source/common/router/config_impl.cc

```diff
@@ -2147,13 +2146,42 @@ PerFilterConfigs::PerFilterConfigs(
     const Protobuf::Map<std::string, ProtobufWkt::Any>& typed_configs,
     const OptionalHttpFilters& optional_http_filters,
     Server::Configuration::ServerFactoryContext& factory_context,
-    ProtobufMessage::ValidationVisitor& validator) {
-  for (const auto& it : typed_configs) {
-    const auto& name = it.first;
-    auto object = createRouteSpecificFilterConfig(name, it.second, optional_http_filters,
-                                                  factory_context, validator);
-    if (object != nullptr) {
-      configs_[name] = std::move(object);
+    ProtobufMessage::ValidationVisitor& validator)
+    : ignore_optional_option_from_hcm_for_route_config_(Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.ignore_optional_option_from_hcm_for_route_config")) {
```

Comment:
Since this is only used within the ctor I think we can avoid making it a class member?

Reply:
good point, right.